onnxruntime/contrib_ops/cpu/bert/attention_base.h

```diff
@@ -68,7 +68,6 @@ class AttentionBase {
                      const Tensor* past_seq_len = nullptr) const;
 
   int num_heads_;                          // number of attention heads
-  int kv_num_heads_;                       // different for k and v for group query attention
```

Comment:
Why is this being removed? Once we de-duplicate `attention_base.h` and `gqa_attention_base.h` to keep only `attention_base.h`, I thought we would like this to be here.

Reply:
A few attributes (like unidirectional, rotary_embedding_dim, qkv_hidden_sizes, past_present_share_buffer) in attention_base is not available in GQA spec:
https://github.com/microsoft/onnxruntime/blob/2684e9717b08a194554445fdbd15d6fe9e14a7ac/onnxruntime/contrib_ops/cpu/bert/attention_base.h#L39-L49

So I decouple gqa_attention_base from attention_base to avoid confusion.